u_parrots.go

```diff
@@ -619,38 +633,86 @@ func (uconn *UConn) generateRandomizedSpec(WithALPN bool) (ClientHelloSpec, erro
 		}
 		p.Extensions = append(p.Extensions, &ks, &pskExchangeModes, &supportedVersionsExt)
 	}
-	err = shuffleTLSExtensions(p.Extensions)
-	if err != nil {
-		return p, err
-	}
+	r.shuffleTLSExtensions(p.Extensions)
 	err = uconn.SetTLSVers(p.TLSVersMin, p.TLSVersMax)
 	if err != nil {
 		return p, err
 	}
 
-	return p, nil
+	return p, r.savedErr
+}
+
+// utlsRandomizer never returns errors, but stores first one in savedErr
+type utlsRandomizer struct {
+	r        io.Reader
+	savedErr error
+}
+
+func makeUtlsRandomizer(seed uint64) *utlsRandomizer {
+	seedBytes := make([]byte, 8)
+	binary.BigEndian.PutUint64(seedBytes, seed)
+	tdHkdf := hkdf.New(sha256.New, seedBytes, []byte("uTLS salty salt"), nil)
```

Comment:
Are all uses of any `utlsRandomizer` guaranteed to read <= 8160 bytes from the entropy source, ever?